### PR TITLE
[All-main] Compose 1.2 snapshot and other dependencies update

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

File: AdvancedStateAndSideEffectsCodelab/app/src/main/java/androidx/compose/samples/crane/base/ExploreSection.kt [hunk 1]
```diff
@@ -114,11 +118,10 @@ private fun ExploreItem(
...
         ExploreImageContainer {
             Box {
-                val painter = rememberImagePainter(
-                    data = item.imageUrl,
-                    builder = {
+                val painter = rememberAsyncImagePainter(
+                    Builder(LocalContext.current).data(data = item.imageUrl).apply(block = fun Builder.() {
                         crossfade(true)
-                    }
+                    }).build()
```
Comment (astamato) [inline AdvancedStateAndSideEffectsCodelab/app/src/main/java/androidx/compose/samples/crane/base/ExploreSection.kt:124]: Have you checked [#801 ](https://github.com/android/compose-samples/pull/801/files) in compose samples? 
```
        val painter = rememberAsyncImagePainter(
            model = ImageRequest.Builder(LocalContext.current)
                .data(item.imageUrl)
                .crossfade(true)
                .build()
        )
```

File: AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt [hunk 2]
```diff
@@ -16,90 +16,32 @@
...
-import androidx.compose.animation.core.infiniteRepeatable
-import androidx.compose.animation.core.keyframes
-import androidx.compose.animation.core.rememberInfiniteTransition
-import androidx.compose.animation.core.spring
-import androidx.compose.animation.core.tween
-import androidx.compose.animation.core.updateTransition
-import androidx.compose.animation.slideInVertically
-import androidx.compose.animation.slideOutVertically
-import androidx.compose.animation.splineBasedDecay
+import androidx.compose.animation.*
```
Comment (simona-anomis) [inline AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt:19]: Wildcard imports in a few places

**Review by simona-anomis — APPROVED** [review]

File: AccessibilityCodelab/build.gradle [hunk 3]
```diff
@@ -48,7 +53,7 @@ subprojects {
             targetExclude("$buildDir/**/*.kt")
             targetExclude('bin/**/*.kt')
 
-            ktlint("0.41.0")
+            ktlint("0.45.2")
```
Comment (simona-anomis) [inline AccessibilityCodelab/build.gradle:56]: Missing userData

**Review by astamato — APPROVED** [review]

File: AccessibilityCodelab/build.gradle (hunk 3 above)
Comment (florina-muntenescu) [reply AccessibilityCodelab/build.gradle:56]: this didn't seem to have user data before either, so i'm leaving it like this.

**Pull request merged**